funcs_utils.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Process\Process;
+
+// THESE FUNCTIONS ARE NOT INTENDED FOR USE IN SCRIPT FILES - only use functions in funcs_scripts.php
+
+/**
+ * Output an error to the console and halt progress
+ * 
+ * error() is not intended for use in scripts specificially because it will halt progress
+ * Instead scripts should be more tolerant though use of functions like check_file_exists() or just
+ * using warning() instead which output a big warning in the console while still continuing
+ */
+function error($message)
+{
+    output_prs_created();
+    output_repos_with_prs_created();
+    io()->error($message);
+    if (!running_unit_tests()) {
+        die;
+    }
+}
+
+/**
+ * Write to a file after trimming the contents and adding a newline
+ */
+function write_file($path, $contents)
+{
+    $contents = trim($contents) . "\n";
+    file_put_contents($path, $contents);
```

Comment:
`write_file()` is also used by `write_file_if_not_exist()` in which case $path won't exist

Reply:
I think, just in case, if in the future we will make some changes and change logic, we should check. Plus I assume it's a good practice to check all passing variable before pass them to the method that can throw an error.

Reply:
Updated